core/src/test/java/org/apache/calcite/rel/type/RelDataTypeFactoryImplTest.java

```diff
@@ -0,0 +1,53 @@
+/*
```

Comment:
The new test cannot fit under SqlTypeFactoryTest?
Can you add also a test in JdbcTest or ReflectiveSchemaTest (or somewhere else) involving IS NULL on structured types which fails without this fix?

Reply:
Tests refactored